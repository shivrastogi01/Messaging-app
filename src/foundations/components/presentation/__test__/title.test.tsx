import React from "react";
import renderWithTheme from "main/testing/render-with-theme";
import {
  Title,
  HeadingVariants,
  HeadingSizes,
  LabelVariants,
  LabelSizes
} from "foundations/components/presentation";

describe("Title", () => {
  test("default matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Title heading="Heading" label="Label"></Title>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("with capitalization matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Title heading="Heading" label="Label" capitalize></Title>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("with heading props matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Title
        heading="Heading"
        label="Label"
        headingProps={{
          variant: HeadingVariants.INVERSE,
          size: HeadingSizes.BIG
        }}
      ></Title>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("with label props matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Title
        heading="Heading"
        label="Label"
        labelProps={{
          variant: LabelVariants.INVERSE,
          size: LabelSizes.SMALL
        }}
      ></Title>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
