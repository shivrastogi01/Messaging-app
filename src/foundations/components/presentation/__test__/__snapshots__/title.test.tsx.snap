// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title default matches the snapshot 1`] = `
.c2 {
  color: inherit;
  font-family: inherit;
  font-size: 15px;
  font-weight: 500;
}

.c3 {
  color: #9B9B9B;
  font-family: inherit;
  font-size: inherit;
  font-weight: inherit;
}

.c0 .c1 {
  margin-bottom: 5px;
}

<div
  class="c0"
>
  <h3
    class="c1 c2"
  >
    Heading
  </h3>
  <span
    class="c3"
  >
    Label
  </span>
</div>
`;

exports[`Title with capitalization matches the snapshot 1`] = `
.c2 {
  color: inherit;
  font-family: inherit;
  font-size: 15px;
  font-weight: 500;
}

.c3 {
  color: #9B9B9B;
  font-family: inherit;
  font-size: inherit;
  font-weight: inherit;
}

.c0 * {
  text-transform: capitalize;
}

.c0 .c1 {
  margin-bottom: 5px;
}

<div
  class="c0"
>
  <h3
    class="c1 c2"
  >
    Heading
  </h3>
  <span
    class="c3"
  >
    Label
  </span>
</div>
`;

exports[`Title with heading props matches the snapshot 1`] = `
.c2 {
  color: inherit;
  font-family: inherit;
  font-size: 15px;
  font-weight: 500;
  color: #FFFFFF;
  font-size: 20px;
  font-weight: 700;
}

.c3 {
  color: #9B9B9B;
  font-family: inherit;
  font-size: inherit;
  font-weight: inherit;
}

.c0 .c1 {
  margin-bottom: 5px;
}

<div
  class="c0"
>
  <h3
    class="c1 c2"
  >
    Heading
  </h3>
  <span
    class="c3"
  >
    Label
  </span>
</div>
`;

exports[`Title with label props matches the snapshot 1`] = `
.c2 {
  color: inherit;
  font-family: inherit;
  font-size: 15px;
  font-weight: 500;
}

.c3 {
  color: #9B9B9B;
  font-family: inherit;
  font-size: inherit;
  font-weight: inherit;
  color: #FFFFFF;
  font-size: 11px;
}

.c0 .c1 {
  margin-bottom: 5px;
}

<div
  class="c0"
>
  <h3
    class="c1 c2"
  >
    Heading
  </h3>
  <span
    class="c3"
  >
    Label
  </span>
</div>
`;
